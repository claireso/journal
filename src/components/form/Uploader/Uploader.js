import React, { useState, useRef } from 'react'
import PropTypes from 'prop-types'

import { create as createThumbnail } from '@services/thumbnails'
import { IconUpload } from '@components/Icons'

import * as S from './Uploader.styles'

const Uploader = (props) => {
  const [{ preview, error }, setState] = useState({ preview: props.preview })
  const input = useRef()

  const handleChange = async (event) => {
    const { accept, onChange } = props

    const files = event.target.files

    if (files.length === 0) return

    const file = files[0]

    if (!file) return

    if (!accept.includes(file.type)) {
      setState({
        preview: null,
        error: 'This file is not allowed'
      })
      input.current.value = ''
      return
    }

    try {
      const thumbnail = await createThumbnail(file)

      setState({ preview: thumbnail, error: null })

      onChange && onChange(thumbnail)
    } catch (err) {
      /* eslint-disable */
      console.error(err)
    }
  }

  return (
    <S.UploaderWrapper>
      {preview && (
        <S.UploaderPreview
          css={{
            background: props.backgroundPreview || '$secondary200'
          }}
        >
          <img src={preview} />
        </S.UploaderPreview>
      )}
      <S.UploaderContent>
        <IconUpload />
        <span>Upload new photo</span>
        <small>(only jpg and png)</small>
        <S.UploaderInput ref={input} type="file" name={props.name} onChange={handleChange} required={props.required} />
      </S.UploaderContent>
      {error && <S.UploaderError>{error}</S.UploaderError>}
    </S.UploaderWrapper>
  )
}

Uploader.propTypes = {
  name: PropTypes.string.isRequired,
  accept: PropTypes.array.isRequired,
  preview: PropTypes.string,
  required: PropTypes.bool,
  onChange: PropTypes.func
}

export default Uploader
