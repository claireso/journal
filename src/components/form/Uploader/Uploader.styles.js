import styled from 'styled-components'

export const UploaderWrapper = styled.div`
  border: 1px solid var(--border-color);
  border-radius: 0.4rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  overflow: hidden;
  position: relative;
`

export const UploaderPreview = styled.div`
  background: ${(props) => props.backgroundColor || 'var(--secondary-normal)'};
  padding: 2rem;

  img {
    display: block;
    max-width: 100%;
    max-height: 17rem;
    margin: 0 auto;
  }
`

export const UploaderContent = styled.div`
  padding: 2rem;

  svg {
    display: block;
    width: 2.4rem;
    height: 2.4rem;
    fill: var(--primary-normal);
    margin: 0 auto;
  }
  span {
    display: block;
    font-size: var(--font-size-normal);
    margin: 1rem 0 0;
    text-align: center;
  }
  small {
    color: var(--border-color-focus);
    display: block;
    font-size: var(--font-size-smaller);
    margin: 0.5rem 0 0;
    text-align: center;
  }
`

export const UploaderInput = styled.input`
  bottom: 0;
  cursor: pointer;
  left: 0;
  opacity: 0;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
`

export const UploaderError = styled.div`
  color: var(--error-normal);
  font-size: var(--font-size-smaller);
  margin: 0.8rem 0 2rem;
  text-align: center;
`
