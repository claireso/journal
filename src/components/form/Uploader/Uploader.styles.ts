import { styled } from '@theme'

export const UploaderWrapper = styled('div', {
  border: '1px solid $gray200',
  borderRadius: '$1',
  boxShadow: '$2',
  overflow: 'hidden',
  position: 'relative'
})

export const UploaderPreview = styled('div', {
  p: '$5',
  img: {
    m: '0 auto',
    display: 'block',
    maxWidth: '100%',
    maxHeight: '17rem'
  }
})

export const UploaderContent = styled('div', {
  p: '$5',
  svg: {
    display: 'block',
    size: '2.4rem',
    fill: '$primary100',
    m: '0 auto'
  },
  span: {
    display: 'block',
    fontSize: '$4',
    mt: '1rem',
    textAlign: 'center'
  },
  small: {
    color: '$gray300',
    display: 'block',
    fontSize: '$3',
    mt: '0.5rem',
    textAlign: 'center'
  }
})

export const UploaderInput = styled('input', {
  bottom: 0,
  cursor: 'pointer',
  left: 0,
  opacity: 0,
  position: 'absolute',
  right: 0,
  top: 0,
  width: '100%'
})

export const UploaderError = styled('div', {
  color: '$error100',
  fontSize: '$3',
  margin: '$2 0 $5',
  textAlign: 'center'
})
