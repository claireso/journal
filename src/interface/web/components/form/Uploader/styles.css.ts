import { recipe } from '@vanilla-extract/recipes'
import { cmpStyles, tokens, p, gap, mx, my, radius } from '@web/theme'

export const wrapper = cmpStyles({
  overflow: 'hidden',
  position: 'relative',
  border: `1px solid ${tokens.colors.neutral.extralight}`,
  boxShadow: tokens.shadows.input,
  ...radius('size-1')
})

export const preview = recipe({
  base: cmpStyles({
    transition: 'opacity 120ms ease-out, background 150ms ease-out',
    ...p('size-5')
  }),
  variants: {
    processing: {
      true: cmpStyles({
        opacity: 0.5
      })
    }
  }
})

export const previewImage = cmpStyles({
  display: 'block',
  maxWidth: '100%',
  maxHeight: '17rem',
  ...mx('size-auto'),
  ...my('size-0')
})

export const content = cmpStyles({
  display: 'flex',
  alignItems: 'center',
  flexDirection: 'column',
  ...p('size-5'),
  ...gap('size-1')
})

export const input = cmpStyles({
  bottom: 0,
  cursor: 'pointer',
  left: 0,
  opacity: 0,
  position: 'absolute',
  right: 0,
  top: 0,
  width: '100%'
})
