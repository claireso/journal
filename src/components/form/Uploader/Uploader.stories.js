import React from 'react'
import { action } from '@storybook/addon-actions'

import Uploader from './index'

const params = {
  title: 'Form/Uploader',
  decorators: [(storyFn) => <div style={{ maxWidth: '500px' }}>{storyFn()}</div>]
}

export default params

export const Basic = () => <Uploader name="media" accept={['image/jpeg', 'image/png']} onChange={action('onChange')} />
