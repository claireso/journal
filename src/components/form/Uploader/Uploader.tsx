import React, { useState, useRef } from 'react'

import { create as createThumbnail } from '@services/thumbnails'
import { IconUpload } from '@components/Icons'

import * as S from './Uploader.styles'

interface UploaderProps {
  name: string
  accept: string[]
  preview?: string
  required?: boolean
  onChange?: (thumbnail: string) => void
  backgroundPreview?: string | null
}

interface State {
  preview?: string | null
  error?: string | null
}

const Uploader = (props: UploaderProps) => {
  const [{ preview, error }, setState] = useState<State>({ preview: props.preview })
  const input = useRef<HTMLInputElement>(null)

  const handleChange = async (event: React.ChangeEvent<HTMLInputElement>) => {
    const { accept, onChange } = props

    const files = event.target.files

    if (!files || files.length === 0) return

    const file = files[0]

    if (!file) return

    if (!accept.includes(file.type)) {
      setState({
        preview: null,
        error: 'This file is not allowed'
      })
      if (input.current) {
        input.current.value = ''
      }
      return
    }

    try {
      const thumbnail = await createThumbnail(file)

      setState({ preview: thumbnail, error: null })

      onChange && onChange(thumbnail)
    } catch (err) {
      /* eslint-disable */
      console.error(err)
    }
  }

  return (
    <S.UploaderWrapper>
      {preview && (
        <S.UploaderPreview
          css={{
            background: props.backgroundPreview || '$secondary200'
          }}
        >
          <img src={preview} />
        </S.UploaderPreview>
      )}
      <S.UploaderContent>
        <IconUpload />
        <span>Upload new photo</span>
        <small>(only jpg and png)</small>
        <S.UploaderInput ref={input} type="file" name={props.name} onChange={handleChange} required={props.required} />
      </S.UploaderContent>
      {error && <S.UploaderError>{error}</S.UploaderError>}
    </S.UploaderWrapper>
  )
}

export default Uploader
