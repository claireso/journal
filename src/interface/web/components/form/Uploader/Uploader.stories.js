import React from 'react'
import { action } from '@storybook/addon-actions'

import Uploader from './index'

const params = {
  title: 'Form/Uploader',
  decorators: [(storyFn) => <div style={{ maxWidth: '500px' }}>{storyFn()}</div>],
  argTypes: { processing: { control: 'boolean' } },
  args: {
    processing: false
  }
}

export default params

export const Basic = (args) => (
  <Uploader
    name="media"
    accept={['image/jpeg', 'image/png']}
    onChange={action('onChange')}
    processing={args.processing}
  />
)
