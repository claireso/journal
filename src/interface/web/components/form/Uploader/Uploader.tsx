import React, { useState, useRef, useCallback } from 'react'

import { create as createThumbnail } from '@web/services/thumbnails'

import { IconUpload } from '@web/components/Icons'
import Spinner from '@web/components/Spinner'

import * as S from './Uploader.styles'

interface UploaderProps {
  name: string
  accept: string[]
  processing: boolean
  preview?: string
  required?: boolean
  previewBackgroundColor?: string | null
  onError: (err: unknown) => void
  onChangeMedia: (file: File) => void
}

const Uploader = ({
  name,
  accept,
  processing,
  onChangeMedia,
  preview,
  required,
  previewBackgroundColor,
  onError
}: UploaderProps) => {
  const [currentPreview, setCurrentPreview] = useState<string | null>(preview ?? null)
  const [error, setError] = useState<string | null>(null)
  const refInput = useRef<HTMLInputElement>(null)

  const onChange = useCallback(
    async (event: React.ChangeEvent<HTMLInputElement>) => {
      const file = event.target.files?.[0]
      if (!file) {
        return
      }

      if (!accept.includes(file.type)) {
        setCurrentPreview(null)
        setError('This file type is not allowed')
        if (refInput.current) {
          refInput.current.value = ''
        }
        return
      }

      setError(null)

      try {
        const thumbnail = await createThumbnail(file)
        setCurrentPreview(thumbnail)
        onChangeMedia(file)
      } catch (err) {
        onError(err)
      }
    },
    [accept, onError, onChangeMedia]
  )

  return (
    <S.UploaderWrapper>
      {currentPreview && (
        <S.UploaderPreview
          css={{
            background: previewBackgroundColor || '$secondary200',
            opacity: processing ? 0.5 : 1
          }}
          data-testid="preview"
        >
          <img src={currentPreview} alt="" />
        </S.UploaderPreview>
      )}
      <S.UploaderContent>
        <>
          <S.UploaderIcon>{processing ? <Spinner /> : <IconUpload />}</S.UploaderIcon>
          <span>Upload new photo</span>
          <small>(only {accept.join(', ')})</small>
        </>

        <S.UploaderInput
          ref={refInput}
          type="file"
          name={name}
          onChange={onChange}
          required={required}
          accept={accept.join(',')}
        />
      </S.UploaderContent>
      {error && <S.UploaderError>{error}</S.UploaderError>}
    </S.UploaderWrapper>
  )
}

export default Uploader
