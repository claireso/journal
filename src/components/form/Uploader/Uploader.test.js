import { render, fireEvent, waitFor } from '@testing-library/react'

import Uploader from './index'

describe('<Uploader />', () => {
  const props = {
    name: 'uploader',
    accept: ['image/jpeg', 'image/png']
  }

  test('should render uploader', () => {
    const { container } = render(<Uploader {...props} />)

    expect(container).toMatchSnapshot()
  })

  test('should render uploader with preview', () => {
    const { container } = render(<Uploader {...props} preview="file.jpg" />)

    expect(container).toMatchSnapshot()
  })

  test('should render error', () => {
    const { container } = render(<Uploader {...props} />)

    const input = container.querySelector('input[type="file"]')

    fireEvent.change(input, {
      target: {
        files: [new File(['(⌐□_□)'], 'mypdf.pdf', { type: 'application/pdf' })]
      }
    })

    expect(container).toMatchSnapshot()
  })

  test('should change file', async () => {
    const spy = jest.fn()
    const { container } = render(<Uploader {...props} onChange={spy} />)

    const input = container.querySelector('input[type="file"]')

    await fireEvent.change(input, {
      target: {
        files: [new File(['(⌐□_□)'], 'mypicture.jpg', { type: 'image/jpeg' })]
      }
    })

    await waitFor(() => {
      expect(container.querySelector('img')).toBeInTheDocument()
      expect(spy).toHaveBeenCalledWith('data:image/jpeg;base64,KOKMkOKWoV/ilqEp')
      expect(container).toMatchSnapshot()
    })
  })
})
