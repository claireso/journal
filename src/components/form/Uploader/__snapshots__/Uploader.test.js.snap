// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Uploader /> should change file 1`] = `
.c0 {
  border: 1px solid var(--border-color);
  border-radius: 0.4rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  overflow: hidden;
  position: relative;
}

.c1 {
  background: var(--secondary-normal);
  padding: 2rem;
}

.c1 img {
  display: block;
  max-width: 100%;
  max-height: 17rem;
  margin: 0 auto;
}

.c2 {
  padding: 2rem;
}

.c2 svg {
  display: block;
  width: 2.4rem;
  height: 2.4rem;
  fill: var(--primary-normal);
  margin: 0 auto;
}

.c2 span {
  display: block;
  font-size: var(--font-size-normal);
  margin: 1rem 0 0;
  text-align: center;
}

.c2 small {
  color: var(--border-color-focus);
  display: block;
  font-size: var(--font-size-smaller);
  margin: 0.5rem 0 0;
  text-align: center;
}

.c3 {
  bottom: 0;
  cursor: pointer;
  left: 0;
  opacity: 0;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        src="data:image/jpeg;base64,KOKMkOKWoV/ilqEp"
      />
    </div>
    <div
      class="c2"
    >
      <svg
        height="16"
        viewBox="0 0 26 28"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM24 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM26 19.5v5c0 0.828-0.672 1.5-1.5 1.5h-23c-0.828 0-1.5-0.672-1.5-1.5v-5c0-0.828 0.672-1.5 1.5-1.5h6.672c0.422 1.156 1.531 2 2.828 2h4c1.297 0 2.406-0.844 2.828-2h6.672c0.828 0 1.5 0.672 1.5 1.5zM20.922 9.375c-0.156 0.375-0.516 0.625-0.922 0.625h-4v7c0 0.547-0.453 1-1 1h-4c-0.547 0-1-0.453-1-1v-7h-4c-0.406 0-0.766-0.25-0.922-0.625-0.156-0.359-0.078-0.797 0.219-1.078l7-7c0.187-0.203 0.453-0.297 0.703-0.297s0.516 0.094 0.703 0.297l7 7c0.297 0.281 0.375 0.719 0.219 1.078z"
        />
      </svg>
      <span>
        Upload new photo
      </span>
      <small>
        (only jpg and png)
      </small>
      <input
        class="c3"
        name="uploader"
        type="file"
      />
    </div>
  </div>
</div>
`;

exports[`<Uploader /> should render error 1`] = `
.c0 {
  border: 1px solid var(--border-color);
  border-radius: 0.4rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  overflow: hidden;
  position: relative;
}

.c1 {
  padding: 2rem;
}

.c1 svg {
  display: block;
  width: 2.4rem;
  height: 2.4rem;
  fill: var(--primary-normal);
  margin: 0 auto;
}

.c1 span {
  display: block;
  font-size: var(--font-size-normal);
  margin: 1rem 0 0;
  text-align: center;
}

.c1 small {
  color: var(--border-color-focus);
  display: block;
  font-size: var(--font-size-smaller);
  margin: 0.5rem 0 0;
  text-align: center;
}

.c2 {
  bottom: 0;
  cursor: pointer;
  left: 0;
  opacity: 0;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
}

.c3 {
  color: var(--error-normal);
  font-size: var(--font-size-smaller);
  margin: 0.8rem 0 2rem;
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        height="16"
        viewBox="0 0 26 28"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM24 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM26 19.5v5c0 0.828-0.672 1.5-1.5 1.5h-23c-0.828 0-1.5-0.672-1.5-1.5v-5c0-0.828 0.672-1.5 1.5-1.5h6.672c0.422 1.156 1.531 2 2.828 2h4c1.297 0 2.406-0.844 2.828-2h6.672c0.828 0 1.5 0.672 1.5 1.5zM20.922 9.375c-0.156 0.375-0.516 0.625-0.922 0.625h-4v7c0 0.547-0.453 1-1 1h-4c-0.547 0-1-0.453-1-1v-7h-4c-0.406 0-0.766-0.25-0.922-0.625-0.156-0.359-0.078-0.797 0.219-1.078l7-7c0.187-0.203 0.453-0.297 0.703-0.297s0.516 0.094 0.703 0.297l7 7c0.297 0.281 0.375 0.719 0.219 1.078z"
        />
      </svg>
      <span>
        Upload new photo
      </span>
      <small>
        (only jpg and png)
      </small>
      <input
        class="c2"
        name="uploader"
        type="file"
      />
    </div>
    <div
      class="c3"
    >
      This file is not allowed
    </div>
  </div>
</div>
`;

exports[`<Uploader /> should render uploader 1`] = `
.c0 {
  border: 1px solid var(--border-color);
  border-radius: 0.4rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  overflow: hidden;
  position: relative;
}

.c1 {
  padding: 2rem;
}

.c1 svg {
  display: block;
  width: 2.4rem;
  height: 2.4rem;
  fill: var(--primary-normal);
  margin: 0 auto;
}

.c1 span {
  display: block;
  font-size: var(--font-size-normal);
  margin: 1rem 0 0;
  text-align: center;
}

.c1 small {
  color: var(--border-color-focus);
  display: block;
  font-size: var(--font-size-smaller);
  margin: 0.5rem 0 0;
  text-align: center;
}

.c2 {
  bottom: 0;
  cursor: pointer;
  left: 0;
  opacity: 0;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <svg
        height="16"
        viewBox="0 0 26 28"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM24 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM26 19.5v5c0 0.828-0.672 1.5-1.5 1.5h-23c-0.828 0-1.5-0.672-1.5-1.5v-5c0-0.828 0.672-1.5 1.5-1.5h6.672c0.422 1.156 1.531 2 2.828 2h4c1.297 0 2.406-0.844 2.828-2h6.672c0.828 0 1.5 0.672 1.5 1.5zM20.922 9.375c-0.156 0.375-0.516 0.625-0.922 0.625h-4v7c0 0.547-0.453 1-1 1h-4c-0.547 0-1-0.453-1-1v-7h-4c-0.406 0-0.766-0.25-0.922-0.625-0.156-0.359-0.078-0.797 0.219-1.078l7-7c0.187-0.203 0.453-0.297 0.703-0.297s0.516 0.094 0.703 0.297l7 7c0.297 0.281 0.375 0.719 0.219 1.078z"
        />
      </svg>
      <span>
        Upload new photo
      </span>
      <small>
        (only jpg and png)
      </small>
      <input
        class="c2"
        name="uploader"
        type="file"
      />
    </div>
  </div>
</div>
`;

exports[`<Uploader /> should render uploader with preview 1`] = `
.c0 {
  border: 1px solid var(--border-color);
  border-radius: 0.4rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  overflow: hidden;
  position: relative;
}

.c1 {
  background: var(--secondary-normal);
  padding: 2rem;
}

.c1 img {
  display: block;
  max-width: 100%;
  max-height: 17rem;
  margin: 0 auto;
}

.c2 {
  padding: 2rem;
}

.c2 svg {
  display: block;
  width: 2.4rem;
  height: 2.4rem;
  fill: var(--primary-normal);
  margin: 0 auto;
}

.c2 span {
  display: block;
  font-size: var(--font-size-normal);
  margin: 1rem 0 0;
  text-align: center;
}

.c2 small {
  color: var(--border-color-focus);
  display: block;
  font-size: var(--font-size-smaller);
  margin: 0.5rem 0 0;
  text-align: center;
}

.c3 {
  bottom: 0;
  cursor: pointer;
  left: 0;
  opacity: 0;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        src="file.jpg"
      />
    </div>
    <div
      class="c2"
    >
      <svg
        height="16"
        viewBox="0 0 26 28"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM24 23c0-0.547-0.453-1-1-1s-1 0.453-1 1 0.453 1 1 1 1-0.453 1-1zM26 19.5v5c0 0.828-0.672 1.5-1.5 1.5h-23c-0.828 0-1.5-0.672-1.5-1.5v-5c0-0.828 0.672-1.5 1.5-1.5h6.672c0.422 1.156 1.531 2 2.828 2h4c1.297 0 2.406-0.844 2.828-2h6.672c0.828 0 1.5 0.672 1.5 1.5zM20.922 9.375c-0.156 0.375-0.516 0.625-0.922 0.625h-4v7c0 0.547-0.453 1-1 1h-4c-0.547 0-1-0.453-1-1v-7h-4c-0.406 0-0.766-0.25-0.922-0.625-0.156-0.359-0.078-0.797 0.219-1.078l7-7c0.187-0.203 0.453-0.297 0.703-0.297s0.516 0.094 0.703 0.297l7 7c0.297 0.281 0.375 0.719 0.219 1.078z"
        />
      </svg>
      <span>
        Upload new photo
      </span>
      <small>
        (only jpg and png)
      </small>
      <input
        class="c3"
        name="uploader"
        type="file"
      />
    </div>
  </div>
</div>
`;
